src/odb/src/db/dbBlock.cpp

```diff
@@ -4097,6 +4097,18 @@ void dbBlock::preExttreeMergeRC(double max_cap, uint corner)
   }
 }
 
+bool dbBlock::designIsRouted()
+{
+  for (dbNet* net : getNets()) {
+    const int pin_count = net->getBTermCount() + net->getITerms().size();
+    if (pin_count > 1 && net->getSWires().empty() && net->getWire() == nullptr
+        && !net->isConnectedByAbutment()) {
```

Comment:
if the SWires is not empty, you should probably verify that they actually have routing in them.
Same for getWire (ensure it actually contains something).

Reply:
I was able to simplify it by getting the wire count of each net. This function already cover both special and non-special nets.